R/msaGaugeRR.R

```diff
@@ -290,85 +289,253 @@ msaGaugeRR <- function(jaspResults, dataset, options, ...){
+
+
+      RRtable2$setData(list(      "source"       = c("Total r & R", "Repeatability", "Reproducibility", "Operator", "Part-To-Part", "Total Variation"),
+                                  "SD"           = SD,
+                                  "studyVar"    = studyVar,
+                                  "percentStudyVar"    = studyVar/max(studyVar) * 100,
+                                  "percentTolerance" = studyVar / options$tolerance * 100))
+
+
+      anovaTables[['RRtable2']] <- RRtable2
+    }
+
+    if (options[["gaugeVarCompGraph"]]) {
+
+      plot <- createJaspPlot(title = "Variation Components Graph", width = 850, height = 500)
+
+      plot$dependOn(c("gaugeVarCompGraph"))
+
+      percentContributionValues <- c(totalRR, repeatability, reproducibility, partToPart)/totalVar * 100
+      SDs <- sqrt(c(totalRR, repeatability, reproducibility, partToPart))
+      studyvars <- SDs * studyVarMultiplier
+      studyVariationValues <- studyvars/max(studyVar) * 100
+      percentToleranceValues <- studyvars / options$tolerance * 100
+
+      plotframe <- data.frame(source = rep(c('Gauge r&R', 'Repeat', 'Reprod', 'Part-to-Part'), 3),
+                              reference = rep(c('Percent Contribution', 'Percent Study Variation', 'Percent Tolerance'), each = 4),
+                              value = c(percentContributionValues, studyVariationValues, percentToleranceValues))
+
+      p <- ggplot2::ggplot() + ggplot2::geom_bar(data = plotframe,
+                                                 mapping = ggplot2::aes(fill =  reference,  y = value, x = source),
+                                                 position="dodge", stat = "identity")
+      p <- jaspGraphs::themeJasp(p) + ggplot2::theme(legend.position = 'right', legend.title = ggplot2::element_blank()) +
+        ggplot2::xlab('') + ggplot2::ylab('Percent') + ggplot2::ylim(c(0, ifelse(max(plotframe['value']) > 100,
+                                                                                 max(plotframe['value'] + 5), 100)))
+
+      plot$plotObject <- p
+
+      anovaTables[['VarCompGraph']] <- plot
+    }
+  }else {
+    RRtable3 <- createJaspTable(title = gettext("r & R Table"))
+    RRtable3$position <- 7
+
+    RRtable3$dependOn(c("gaugeANOVA", "operators", "parts", "measurements"))
+
+    RRtable3$addColumnInfo(name = "Source", title = gettext("Source"), type = "string")
+    RRtable3$addColumnInfo(name = "Variation", title = gettext("Variation"), type = "number")
+    RRtable3$addColumnInfo(name = "Percent", title = gettext("Percent Contribution to Total"), type = "number")
+
+    RRtable3$setData(list(      "Source"       = c("Total r & R", "Repeatability", "Reproducibility", "Operator", "Part-To-Part", "Operator x Part", "Total Variation")))
+
+    anovaTables[['RRtable3']] <- RRtable3
+
+    if(ready){
+      RRtable3$setError(gettextf(paste("Number of observations is < 2 in ", measurements, "after grouping on ", parts,",",operators)))
```

Comment:
This won't work for translations! You need something like this, but that does assumes that `parts` and `operators` have length 1. If they don't try something like `paste(parts, collapse = ", ")`.
```suggestion
      RRtable3$setError(gettextf("Number of observations is < 2 in %s after grouping on %s", parts, operators))
```